megalinter/linters/GitleaksLinter.py

```diff
@@ -13,48 +13,47 @@
 
 class GitleaksLinter(Linter):
     def __init__(self, params=None, linter_config=None):
-        self.is_pr, self.pr_source_sha, self.pr_target_sha = self.get_pr_data()
         super().__init__(params, linter_config)
+        self.only_pr_commits = config.get("REPOSITORY_GITLEAKS_ONLY_PR_COMMITS", "false")
+        if self.only_pr_commits == "true":
+            self.is_pr, self.pr_source_sha, self.pr_target_sha = self.get_pr_data()
+
 
     def get_pr_data(self):
```

Comment:
I'm pretty sure that such code could be reusable elsewhere by other linters, reporters or even core stuff, maybe you could put it in a new utils class ?

Reply:
@nvuillam That was my initial thought and looked for utils class :) and realized it does not exist, and only utils for reporters is. Sure, will move the code.